packages/shared/src/utils/structuredClone.ts

```diff
@@ -0,0 +1,10 @@
+// Attach the polyfill as a Global function
+import structuredClone from '@ungap/structured-clone';
+
+if (!('structuredClone' in globalThis)) {
+  globalThis.structuredClone = ((
+    value: StructuredSerializeOptions | undefined,
+    options?: { lossy?: boolean | undefined } | undefined,
+  ): StructuredSerializeOptions | undefined =>
+    structuredClone(value, options)) as any;
```

Comment:
does this work?

`globalThis.structuredClone = structuredClone as any`

Reply:
- [ ]  I have tested it locally and it seems to do its job just fine when it comes to cloning a (nested) object as seperate instance
- [ ] I have played with `if (!('structuredClone' in globalThis)) {`
...replacing 'structuredClone' with non-existent window objects. This adds this function to it
- [ ] in regards of structuredClone on my browser, there was no change and it still uses the native structuredClone function. So it seems to do what a fallback should do and only uses this polyfill when needed
- [ ] @ilijabojanovic  What I have not tested is the actualy behavior on browsers which do not support it: Which is the reason for this ticket. We need  to check that on old browsers --> https://caniuse.com/?search=structuredClone

![image](https://github.com/lifinance/jumper.exchange/assets/43956540/e2fb0d60-bcb6-4005-9384-72897c8f6ee8)

Reply:
I mean, we should probably make it as one-liner without complicated types because it's a polyfill anyway 🙂

`globalThis.structuredClone = structuredClone as any`

Reply:
Makes sense, I simplified it

Reply:
@dennyscode, why not simplify it even further? 😅